src/subdataframe/subdataframe.jl

```diff
@@ -51,14 +51,14 @@ immutable SubDataFrame{T <: AbstractVector{Int}} <: AbstractDataFrame
     parent::DataFrame
     rows::T # maps from subdf row indexes to parent row indexes
 
-    @compat function SubDataFrame{T}(parent::DataFrame, rows::T) where T
+    function (::Type{SubDataFrame{T}}){T}(parent::DataFrame, rows::T)
```

Comment:
I'm sure it is recommended but it works without warnings across 0.5 and 0.6 so I think it the right solution until we drop 0.5 support.